File: 1301-1400/1338-Reduce Array Size to The Half/1338-Reduce Array Size to The Half.cpp
class Solution {
public:
    int minSetSize(vector<int>& arr) {
        unordered_map<int, int> table;
        for (int a : arr) {
            ++table[a];
        }
        
        multiset<int> count;
        for (auto p : table) {
            count.insert(p.second);
        }
        int minSize = 0, remove = 0;
        for (auto it = count.rbegin(); remove * 2 < arr.size(); ++it) {
            ++minSize;
            remove += *it;
        }
        return minSize;
    }
};

File: 1301-1400/1338-Reduce Array Size to The Half/1338-Reduce Array Size to The Half.py
import collections
class Solution:
    def minSetSize(self, arr: List[int]) -> int:
        table = collections.Counter(arr)
        minSize = count = 0
        for k in sorted(table, key=lambda x: table[x], reverse=True):
            count += table[k]
            minSize += 1
            if count >= len(arr) // 2:
                break
        return minSize
